conf/gulp-tasks/bundle/bundletaskfactory.js

```diff
@@ -35,16 +26,47 @@ class BundleTaskFactory {
    * @returns {Function} Gulp task for producing the requested SDK bundle.
    */
   create (bundleType) {
+    let bundleFunction;
     switch (bundleType) {
       case BundleType.MODERN:
-        return () => this._modernBundle();
+        bundleFunction = (callback) => this._modernBundle(callback);
+        break;
       case BundleType.LEGACY_IIFE:
-        return () => this._legacyBundleIIFE();
+        bundleFunction = (callback) => this._legacyBundleIIFE(callback);
+        break;
       case BundleType.LEGACY_UMD:
-        return () => this._legacyBundleUMD();
+        bundleFunction = (callback) => this._legacyBundleUMD(callback);
+        break;
       default:
         throw new Error('Unrecognized BundleType');
     }
+    Object.defineProperty(bundleFunction, 'name', { value: this._getBundleName(bundleType) });
+    return bundleFunction;
+  }
+
+  /**
+   * Provides a Gulp task to minify an SDK bundle of the specified type.
+   *
+   * @param {BundleType} bundleType The type of SDK bundle to minify.
+   * @returns {Function} Gulp task for minifying the requested SDK bundle.
+   */
+  minify (bundleType) {
```

Comment:
I think this method gives the Factory two responsibilities. It provides bundling tasks and minifying tasks. It doesn't seem like you can mix and match bundlers and minifiers. Given that that's the case, would it make more sense for this factory to return a Gulp task that both bundles & minifies a `bundleType`? If we make that change, probably worth thinking about if this Factory needs renaming.

Reply:
Yeah, I was thinking that as I was writing this! Bundling and minifying are similar because they both are gulp tasks and use the "this._localePrefix" and "BundleType", but otherwise they are distinct things. 

I originally had the "create" do both bundling and minifying, but our dev task currently doesn't minify. Since minifying adds time and it'd be nice to keep the time to rebuild on `npm run dev` as fast as possible, it seemed like keeping the tasks separate was a better option. I can split the bundle minifying into a separate class, like a `MinifyTaskFactory`?

Reply:
Added a MinifyTaskFactory class!